### PR TITLE
DrivAerML: Fourier+no-EMA+T_max=30 long training replication (validate winner)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current DrivAerML best (51.35%, PR #2475) was only 2 epochs — the run was clearly still descending at cutoff. historia (PR #2499) is also running long training with this config. This is an independent replication with the identical winning config to: (1) validate that long training converges well, (2) establish error bars on convergence, and (3) compare results with historia's concurrent run. If both replicas converge to similar values, we have high confidence in the result; divergence would indicate training instability.

## Instructions

Run the exact winning config from PR #2475, but for the full 180-min budget. Use `--epochs 999` — the default of 2 epochs is a known bug.

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group violet-drivaerml-replica
```

## Baseline

- **Current best (DrivAerML):** val_primary/surface_rel_l2_pct = 51.35% (val) / 52.06% (test)
- **Best PR:** #2475 (chihiro — Fourier + no-EMA, T_max=30, lr=5e-4, 2 epochs only)
- **W&B run:** 5ncrjm32 (winner, T_max=30)
- **Concurrent replication:** historia PR #2499 (running same config)
- **External target:** <3.71% (AB-UPT, ~500 epochs)

## Reporting

Please report:
- `val_primary/surface_rel_l2_pct` at each epoch (full per-epoch trajectory)
- Final val and test `surface_rel_l2_pct`
- Number of epochs completed in 180 min
- W&B run ID
- Whether results converged or were still descending at cutoff
- Comparison with historia's concurrent run (if results available)

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT violet:

## Results

### DrivAerML: Fourier+no-EMA+T_max=30 long training replication

**Config:** Exact winning config from PR #2475 with extended training (--epochs 999, 30-min timeout).

| Metric | Baseline (PR #2475, 2 ep) | This run (6 ep) | Change |
|--------|---------------------------|------------------|--------|
| val_primary/surface_rel_l2_pct | 51.35% | **33.65%** | **-34.5% relative** |
| test_primary/surface_rel_l2_pct | 52.06% | **34.00%** | **-34.7% relative** |

### Per-epoch trajectory

| Epoch | val_primary/surface_rel_l2_pct | Loss | LR |
|-------|-------------------------------|------|----|
| 1 | 59.71% | 0.789 | 2.16e-5 |
| 2 | 51.98% | 0.535 | 4.17e-4 |
| 3 | 43.83% | 0.409 | 1.73e-4 |
| 4 | 40.89% | 0.324 | 2.24e-4 |
| 5 | 35.84% | 0.274 | 3.75e-4 |
| 6 | 33.65% | 0.240 | 4.77e-5 |

### Run details

- **W&B run:** xm765o85
- **Epochs completed:** 6 (30-min timeout)
- **Command:**
  ```bash
  cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
    --dataset drivaerml --optimizer adamw --lr 5e-4 \
    --cosine-t-max 30 --no-use-ema --enable-fourier \
    --epochs 999 --batch-size 1 \
    --drivaerml-train-surface-points 50000 \
    --drivaerml-eval-surface-points 50000 \
    --max-train-batches 394 --max-eval-batches 200 \
    --wandb-group violet-drivaerml-replica \
    --agent violet --wandb-name "violet/drivaerml-fourier-noema-replica"
  ```

### What happened

Long training confirms massive headroom in the Fourier+no-EMA+T_max=30 config. Going from 2→6 epochs delivered a **34.5% relative improvement** (51.35% → 33.65% val). The model was **still descending at cutoff** — loss dropped monotonically from 0.789 to 0.240, and val metric improved every single epoch with no sign of plateauing.

Key observations:
- **Epoch 2 replicates the baseline closely** (51.98% vs 51.35%), confirming reproducibility of the original PR #2475 result.
- **Cosine restart dynamics are visible**: LR drops to ~2.16e-5 at epoch 1 (trough), peaks at ~4.17e-4 at epoch 2, and completes another cycle by epoch 6 (4.77e-5). Each restart cycle delivers fresh descent.
- **Still strongly improving** — the epoch 5→6 drop (35.84% → 33.65%) is still 2.2 absolute points. More epochs would almost certainly push below 30%.
- **Test tracks val closely** (34.00% test vs 33.65% val), indicating no overfitting.

### Comparison with historia (PR #2499)

Historia's concurrent replication has not posted results yet. Will update if results become available.

### Suggested follow-ups

1. **More epochs** — With 180-min budget (vs our 30-min), we'd expect ~18+ epochs and potentially sub-25% val. This is the single highest-leverage next step for DrivAerML.
2. **T_max tuning at longer training** — T_max=30 means a restart every ~6 epochs at 394 steps/epoch. At longer training, T_max=50 or T_max=60 might allow deeper per-cycle descent.
3. **Larger model capacity** — 4L/256d or 5L/256d variants may benefit more from extended training.

**Comment by tcapelle** [comment]
ADVISOR: Merging this PR — excellent replica result! val_primary/surface_rel_l2_pct = 33.65% (W&B: xm765o85), a 34.5% relative improvement over the 51.35% baseline. The result is clean (val≈test, 34.00%), monotonically decreasing, and still converging at epoch 6. This is the best DrivAerML result to date. Critical insight confirmed: the original 2-epoch baseline was severely training-time limited.
